src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java

```diff
@@ -126,13 +127,15 @@ private FindPathResult findPath(AvailableNetwork network, SwitchId startSwitchId
         FindOneDirectionPathResult pathFindResult = getPath.get();
         List<Edge> forwardPath = pathFindResult.getFoundPath();
         if (forwardPath.isEmpty()) {
-            throw new UnroutableFlowException(format("Can't find a path from %s to %s", start, end));
+            throw new UnroutableFlowException(format("Can't find a path from %s to %s. Reasons: %s",
+                    start, end, StringUtils.join(pathFindResult.getPathNotFoundReasons(), ", ")));
```

Comment:
Just a question: can getPathNotFoundReasons() be null?

Reply:
I think that here it can't be null, but I feel safer checking that it actually is not null. Thanks

Reply:
done